src/script/miniscript.h

```diff
@@ -1194,7 +1194,7 @@ struct Node {
                 case Fragment::OR_I:
                     return subs[0] || subs[1];
                 case Fragment::THRESH:
-                    return std::count(subs.begin(), subs.end(), true) >= node.k;
+                    return unsigned(std::count(subs.begin(), subs.end(), true)) >= node.k;
```

Comment:
Maybe best to avoid C-style casts?

```suggestion
                    return static_cast<uint32_t>(std::count(subs.begin(), subs.end(), true)) >= node.k;
```